ktor-http/src/io/ktor/http/URLBuilder.kt

```diff
@@ -38,7 +36,7 @@ class URLBuilder(
         }
         out.append(host)
 
-        if (port != protocol.defaultPort) {
```

Comment:
To remain as backwards compatible with previous behavior as possible, I left the check to only add if the port is not the protocol's default port